examples: Add bmi270 sample application without SCU

shino-ixy opened the pull request:
I've written User Mode Driver for BMI270(without SCU), and create sample application that can visible tilt of SPRESENSE Main Board by LED[0-3].

SPRESENSE commented on sdk/configs/examples/bmi270_no_scu/defconfig:
```diff
@@ -0,0 +1,9 @@
++CXD56_I2C0_SCUSEQ=n
```
Create a `defconfig` using `mkdefconfig.py`.
```
$ cd spresense/sdk
$ ./tools/mkdefconfig.py examples/bmi270_no_scu
```

shino-ixy replied:
Thanks, teaching usage of tool. I success to generate defconfig

SPRESENSE commented on examples/bmi270_no_scu/Kconfig:
```diff
@@ -0,0 +1,34 @@
+	int "Sensor stack size"
+	default 2048
+
+config EXAMPLES_BMI270_NO_SCU_DEBUFON
```
```suggestion
config EXAMPLES_BMI270_NO_SCU_DEBUGON
```

shino-ixy replied:
as you said. Thank you.

SPRESENSE commented on examples/bmi270_no_scu/Kconfig:
```diff
@@ -0,0 +1,34 @@
+	default 2048
+
+config EXAMPLES_BMI270_NO_SCU_DEBUFON
+	tristate "Enable Debug Print"
```
```suggestion
	bool "Enable Debug Print"
```

shino-ixy replied:
as you said. Thank you.

SPRESENSE reviewed (changes requested):
@shino-ixy Thank you for your contribution!

* Please change the target branch from `master` to `develop`.
(See https://github.com/sonydevworld/spresense/blob/master/CONTRIBUTING.md)

* Please remove some `.DS_Store` files.
I will add this filename into `.gitignore`.

* commit message
The prefix before the colon `:` indicates the context.
Referring to other commits, change it and PR title as below.
"examples: Add bmi270 sample application without SCU"

* coding style
Please change your coding style to match the NuttX's nxstyle. You can use the `checkpatch.sh` tool. e.g.
```
$ cd spresense/sdk
$ ../nuttx/tools/checkpatch.sh -f <filename>
```

* License
 * For files without a license description, please attach an Apache 2.0 license.
(Refer to https://github.com/sonydevworld/spresense-nuttx-apps/blob/new-master/examples/scd41/scd41_main.c#L1-#L19)
 * It seems an `i2c_bmi270.c` refers to the original Bosch license.
Please be sure to grant the appropriate license.
If you have mixed licenses, please separate files for each license.

Let me review the rest after the revised pull-request is issued.

shino-ixy commented:
@SPRESENSE 
Thank you for Reviewing!!

> Please change the target branch from master to develop.
> commit message
> 
OK, I'd changed.

> Please remove some .DS_Store files.
Sorry, I'd deleted these files.

> coding style
Thank you for attention. 
I've updated each codes.

> License
I've separeted codes of BMI270 config table for separating License,
and insert Licesnse each files.

Please re-Review this pull request.

SPRESENSE reviewed (changes requested):
Let me suggest changing the application name from `bmi270_no_scu` to `i2c_direct`.

As you know, the bmi270 driver should originally be created as a sensor driver.
However, it is also possible to control `/dev/i2c` directly, as this example application.

I think it would be appropriate to use the generic name `i2c_direct` as a sample for controlling an i2c character device. Initially only the bmi270 will exist, but we would like to be able to add devices in the future.

And,  please rebase as it contains a merge commit.

shino-ixy commented:
***@***.***> @sonydevworld/spresense 

Thank you for reviewing.

> Let me suggest changing the application name from bmi270_no_scu to i2c_direct.

That’s right! I’ve renamed files and definitions in codes.

And I’ve updated codes to follow each comment, applied rabasing “develop”.

 

From: SPRESENSE ***@***.***> 
Sent: Tuesday, September 20, 2022 5:11 PM
To: sonydevworld/spresense ***@***.***>
Cc: Shino - Ixy Design Studio ***@***.***>; Mention ***@***.***>
Subject: Re: [sonydevworld/spresense] examples: Add bmi270 sample application without SCU (PR #487)

 

@SPRESENSE requested changes on this pull request.

Let me suggest changing the application name from bmi270_no_scu to i2c_direct.

As you know, the bmi270 driver should originally be created as a sensor driver.
However, it is also possible to control /dev/i2c directly, as this example application.

I think it would be appropriate to use the generic name i2c_direct as a sample for controlling an i2c character device. Initially only the bmi270 will exist, but we would like to be able to add devices in the future.

And, please rebase as it contains a merge commit.

  _____  

In examples/bmi270_no_scu/include/i2c_common.h <https://github.com/sonydevworld/spresense/pull/487#discussion_r974935513> :

> +
 
+typedef struct _i2c_ctrl_type
 
+{
 
+  int fd;
 
+  int i2c_addr;
 
+  int speed;
 
+} i2c_ctrl_t;
 
+
 
+int i2c_reg_write(i2c_ctrl_t *pi2c, uint8_t reg, uint8_t value);
 
+int i2c_reg_write_burst(i2c_ctrl_t *pi2c,
 
+  uint8_t reg, uint8_t *pvalue, int len);
 
+int i2c_reg_read(i2c_ctrl_t *pi2c,
 
+  uint8_t reg, uint8_t *value, int16_t len);
 
+void us_sleep(int us);
 
+
 
+#endif /* _I2C_COMMON_H_ */
 

Please fix "No newline at end of file".

  _____  

In examples/bmi270_no_scu/include/i2c_bmi270.h <https://github.com/sonydevworld/spresense/pull/487#discussion_r974935601> :

> +  int gyr_table_pos;
 
+  axis_t *acc_table;
 
+  axis_t *gyr_table;
 
+} i2c_bmi270_t;
 
+
 
+int init_bmi270(i2c_bmi270_t *pctrl);
 
+int exec_dequeue_fifo(i2c_bmi270_t *pctrl);
 
+int get_latest_acc(axis_t *pd, i2c_bmi270_t *pctrl);
 
+int get_latest_gyr(axis_t *pd, i2c_bmi270_t *pctrl);
 
+
 
+/* bmi270.c */
 
+
 
+int get_bmi270_config_file_size(void);
 
+uint8_t *get_bmi270_config_file_addr(void);
 
+
 
+#endif /* _I2C_BMI270_H_ */
 

Please fix "No newline at end of file".

  _____  

In examples/bmi270_no_scu/include/debug_printf.h <https://github.com/sonydevworld/spresense/pull/487#discussion_r974935652> :

> +#define DPRINT_ERROR(...)  \
 
+{ \
 
+printf("ERROR: %s L#%d ", __func__, __LINE__); \
 
+printf(__VA_ARGS__); \
 
+printf("\n"); \
 
+}
 
+#else
 
+#define DPRINT_ERROR(...)
 
+#endif
 
+
 
+#ifdef __cplusplus
 
+}
 
+
 
+#endif
 
+
 
+#endif // _DEBUG_PRINT_H
 

Please fix "No newline at end of file".

  _____  

In examples/bmi270_no_scu/bmi270_no_scu_main.c <https://github.com/sonydevworld/spresense/pull/487#discussion_r974936736> :

> @@ -0,0 +1,288 @@
 
+/**
 
+ * examples/bmi270_no_scu/bmi270_no_scu_main.c
 
+ *
 
+ *  Copyright (C) 2022-202, Ixy Design Studio, All Rights Reserved
 

Is this typo?

⬇️ Suggested change 

- *  Copyright (C) 2022-202, Ixy Design Studio, All Rights Reserved
 
+ *  Copyright (C) 2022, Ixy Design Studio, All Rights Reserved
 
  _____  


In examples/bmi270_no_scu/i2c_bmi270.c <https://github.com/sonydevworld/spresense/pull/487#discussion_r974936986> :

> @@ -0,0 +1,483 @@
 
+/**
 
+ * examples/bmi270_no_scu/i2c_bmi270.c
 
+ *
 
+ *  Copyright (C) 2022-202, Ixy Design Studio, All Rights Reserved
 

ditto

⬇️ Suggested change 

- *  Copyright (C) 2022-202, Ixy Design Studio, All Rights Reserved
 
+ *  Copyright (C) 2022, Ixy Design Studio, All Rights Reserved
 
  _____  


In examples/bmi270_no_scu/i2c_common.c <https://github.com/sonydevworld/spresense/pull/487#discussion_r974937444> :

> @@ -0,0 +1,182 @@
 
+/**
 
+ * examples/bmi270_no_scu/i2c_common.c
 
+ *
 
+ *  Copyright (C) 2022-202, Ixy Design Studio, All Rights Reserved
 

ditto

⬇️ Suggested change 

- *  Copyright (C) 2022-202, Ixy Design Studio, All Rights Reserved
 
+ *  Copyright (C) 2022, Ixy Design Studio, All Rights Reserved
 
  _____  


In examples/bmi270_no_scu/include/debug_printf.h <https://github.com/sonydevworld/spresense/pull/487#discussion_r974950131> :

> + * Copyright 2010-2015 Amazon.com, Inc.
 
+ * or its affiliates. All Rights Reserved.
 
+ *
 
+ * Licensed under the Apache License, Version 2.0 (the "License").
 
+ * You may not use this file except in compliance with the License.
 
+ * A copy of the License is located at
 
+ *
 
+ *  http://aws.amazon.com/apache2.0
 
+ *
 
+ * or in the "license" file accompanying this file. This file is distributed
 
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 
+ * express or implied. See the License for the specific language governing
 
+ * permissions and limitations under the License.
 

Let me check if the copyright is correct. Do you bring this file from Amazon?

  _____  

In examples/bmi270_no_scu/include/i2c_bmi270.h <https://github.com/sonydevworld/spresense/pull/487#discussion_r974962327> :

> @@ -0,0 +1,83 @@
 
+/**
 
+ * examples/bmi270_no_scu/include/i2c_bmi270.h
 
+ *
 
+ *  Copyright (C) 2022-202, Ixy Design Studio, All Rights Reserved
 

ditto

⬇️ Suggested change 

- *  Copyright (C) 2022-202, Ixy Design Studio, All Rights Reserved
 
+ *  Copyright (C) 2022, Ixy Design Studio, All Rights Reserved
 
  _____  


In examples/bmi270_no_scu/include/i2c_common.h <https://github.com/sonydevworld/spresense/pull/487#discussion_r974962575> :

> @@ -0,0 +1,56 @@
 
+/**
 
+ * examples/bmi270_no_scu/include/i2c_common.h
 
+ *
 
+ *  Copyright (C) 2022-202, Ixy Design Studio, All Rights Reserved
 

ditto

⬇️ Suggested change 

- *  Copyright (C) 2022-202, Ixy Design Studio, All Rights Reserved
 
+ *  Copyright (C) 2022, Ixy Design Studio, All Rights Reserved
 
  _____  


In sdk/configs/examples/bmi270_no_scu/defconfig <https://github.com/sonydevworld/spresense/pull/487#discussion_r974964538> :

> @@ -0,0 +1,5 @@
 
+-CXD56_I2C0_SCUSEQ=n
 
+-CXD56_SPI3_SCUSEQ=n
 
++CXD56_SCU=n
 
++EXAMPLES_BMI270_NO_SCU=y
 
++HOST_MACOS=y
 

Please remove a line of +HOST_MACOS=y.

⬇️ Suggested change 

-+HOST_MACOS=y
 

I will modify mkdefconfig.py so that this line does not appear.

  _____  

In examples/bmi270_no_scu/i2c_bmi270.c <https://github.com/sonydevworld/spresense/pull/487#discussion_r974966416> :

> +#define BMI270_REG_NV_CONF      (0x70)
 
+#define BMI270_REG_OFFSET(n)    (0x71 + n)
 
+#define BMI270_REG_PWR_CONF     (0x7c)
 
+#define BMI270_REG_PWR_CTRL     (0x7d)
 
+#define BMI270_REG_CMD          (0x7e)
 
+
 
+#define BMI270_STORE_TABLE_LENGTH (1024)
 
+#define BMI270_FIFO_MAX_LENGTH  (2560)
 
+
 
+/****************************************************************************
 
+ * Private Function Prototypes
 
+ ****************************************************************************/
 
+
 
+const uint8_t bmi270_config_file[];
 
+static int enable_fifo_bmi270(i2c_ctrl_t *pi2c);
 
+static void bcm270_fifo_decoder(i2c_bmi270_t *pctrl);
 

⬇️ Suggested change 

-static void bcm270_fifo_decoder(i2c_bmi270_t *pctrl);
 
+static void bmi270_fifo_decoder(i2c_bmi270_t *pctrl);
 
  _____  


In examples/bmi270_no_scu/i2c_bmi270.c <https://github.com/sonydevworld/spresense/pull/487#discussion_r974966952> :

> +static void
 
+bcm270_fifo_decoder
 
+(i2c_bmi270_t *pctrl)
 

⬇️ Suggested change 

-static void
 
-bcm270_fifo_decoder
 
-(i2c_bmi270_t *pctrl)
 
+static void bmi270_fifo_decoder(i2c_bmi270_t *pctrl)
 
  _____  


In examples/bmi270_no_scu/i2c_bmi270.c <https://github.com/sonydevworld/spresense/pull/487#discussion_r974967150> :

> +  if (ret < 0)
 
+    {
 
+      return -1;
 
+    }
 
+
 
+  pctrl->fifo_depth = CONV(fifo_len_reg, 0);
 
+  DPRINT_DEBUG("FIFO_LENGTH=%d", pctrl->fifo_depth);
 
+
 
+  ret = i2c_reg_read(pi2c, BMI270_REG_FIFO_DATA,
 
+  pctrl->fifo, pctrl->fifo_depth);
 
+  if (ret < 0)
 
+    {
 
+      return -1;
 
+    }
 
+
 
+  bcm270_fifo_decoder(pctrl);
 

⬇️ Suggested change 

-  bcm270_fifo_decoder(pctrl);
 
+  bmi270_fifo_decoder(pctrl);
 
  _____  


In examples/bmi270_no_scu/include/debug_printf.h <https://github.com/sonydevworld/spresense/pull/487#discussion_r974971039> :

> +#ifdef CONFIG_EXAMPLES_BMI270_NO_SCU_DEBUG_ON
 
+#define ENABLE_DPRINT_DEBUG
 
+#define ENABLE_DPRINT_TRACE
 
+#define ENABLE_DPRINT_INFO
 
+#define ENABLE_DPRINT_WARN
 
+#define ENABLE_DPRINT_ERROR
 
+#endif
 

Individual debug levels such as TRACE, INFO, WARN and ERROR should be able to be configured separately in Kconfig.

  _____  

In examples/bmi270_no_scu/bmi270_no_scu_main.c <https://github.com/sonydevworld/spresense/pull/487#discussion_r974976429> :

> +    }
 
+
 
+  bmi270.i2c.fd = fd;
 
+
 
+  /** init bmi270 */
 
+
 
+  ret = init_bmi270(&bmi270);
 
+  if (ret < 0)
 
+    {
 
+      printf("ERROR: Failed to initialize: %d\n", ret);
 
+      return -1;
 
+    }
 
+
 
+  /* -- WAIT 250ms -- */
 
+
 
+  us_sleep(250 * 1000); /* 250ms */
 

I think it would be better to use usleep directly instead of wrapper like us_sleep, what do you think?

  _____  

In examples/bmi270_no_scu/i2c_bmi270.c <https://github.com/sonydevworld/spresense/pull/487#discussion_r974980342> :

> + *
 
+ * @param fd file descriptor
 
+ * @return int success == 0
 
+ */
 
+
 
+int
 
+init_bmi270
 
+(i2c_bmi270_t *pctrl)
 
+{
 
+  int ret;
 
+  uint8_t chipid;
 
+  i2c_ctrl_t *pi2c = &pctrl->i2c;
 
+
 
+  /* alloc working memory */
 
+
 
+  pctrl->fifo = (uint8_t *)malloc(BMI270_FIFO_MAX_LENGTH);
 

Memory leak. This free does not exist in normal case when the application is terminated.

  _____  

In examples/bmi270_no_scu/i2c_bmi270.c <https://github.com/sonydevworld/spresense/pull/487#discussion_r974980436> :

> +{
 
+  int ret;
 
+  uint8_t chipid;
 
+  i2c_ctrl_t *pi2c = &pctrl->i2c;
 
+
 
+  /* alloc working memory */
 
+
 
+  pctrl->fifo = (uint8_t *)malloc(BMI270_FIFO_MAX_LENGTH);
 
+  if (pctrl->fifo == NULL)
 
+    {
 
+      return -1;
 
+    }
 
+
 
+  /* alloc store memory for ACCEL */
 
+
 
+  pctrl->acc_table = (uint8_t *)malloc(BMI270_STORE_TABLE_LENGTH);
 

Memory leak. This free does not exist in normal case when the application is terminated.

  _____  

In examples/bmi270_no_scu/i2c_bmi270.c <https://github.com/sonydevworld/spresense/pull/487#discussion_r974980511> :

> +    }
 
+
 
+  /* alloc store memory for ACCEL */
 
+
 
+  pctrl->acc_table = (uint8_t *)malloc(BMI270_STORE_TABLE_LENGTH);
 
+  if (pctrl->acc_table == NULL)
 
+    {
 
+      free(pctrl->fifo);
 
+      return -1;
 
+    }
 
+
 
+  pctrl->acc_table_pos = 0;
 
+
 
+  /* alloc store memory for GYRO */
 
+
 
+  pctrl->gyr_table = (uint8_t *)malloc(BMI270_STORE_TABLE_LENGTH);
 

Memory leak. This free does not exist in normal case when the application is terminated.

  _____  

In sdk/configs/examples/bmi270_no_scu/defconfig <https://github.com/sonydevworld/spresense/pull/487#discussion_r974996413> :

> +-CXD56_I2C0_SCUSEQ=n
 
+-CXD56_SPI3_SCUSEQ=n
 
++CXD56_SCU=n
 

These 3 lines seem unnecessary.

*	CXD56_I2C0_SCUSEQ must be disabled, but it is disabled by default.
*	SPI3 is not relevant to this example.
*	It is not necessary to disable CXD56_SCU.

⬇️ Suggested change 

--CXD56_I2C0_SCUSEQ=n
 
--CXD56_SPI3_SCUSEQ=n
 
-+CXD56_SCU=n
 
+-CXD56_I2C0_SCUSEQ=n
 
+-CXD56_SPI3_SCUSEQ=n
 
++CXD56_SCU=n
 
  _____  


In examples/bmi270_no_scu/include/i2c_bmi270.h <https://github.com/sonydevworld/spresense/pull/487#discussion_r974998752> :

> +#include <errno.h>
 
+#include <sys/ioctl.h>
 
+#include <nuttx/i2c/i2c_master.h>
 
+#include <nuttx/arch.h>
 
+#include <arch/board/board.h>
 
+#include <arch/chip/pin.h>
 
+#include "i2c_common.h"
 
+#include "debug_printf.h"
 
+
 
+/****************************************************************************
 
+ * Pre-processor Definitions
 
+ ****************************************************************************/
 
+
 
+#define I2C_DEVNAME_FOR_BMI270  "/dev/i2c0"
 
+#define BMI270_I2C_ADDRESS  (0x69)
 
+#define I2C_SPEED           (1000*1000) /* fast mode plus */
 

Fast mode+ is not supported.

⬇️ Suggested change 

-#define I2C_SPEED           (1000*1000) /* fast mode plus */
 
+#define I2C_SPEED           (400*1000) /* fast mode */
 

—
Reply to this email directly, view it on GitHub <https://github.com/sonydevworld/spresense/pull/487#pullrequestreview-1113211844> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AYGIOTVAQOAHQOFVIBO5LNLV7FWQZANCNFSM6AAAAAAQLCDAZ4> .
You are receiving this because you were mentioned.  <https://github.com/notifications/beacon/AYGIOTUPHR7J7J2PZLJHY4TV7FWQZA5CNFSM6AAAAAAQLCDAZ6WGG33NNVSW45C7OR4XAZNRKB2WY3CSMVYXKZLTORJGK5TJMV32UY3PNVWWK3TUL5UWJTSCLJB4I.gif> Message ID: ***@***.*** ***@***.***> >

SPRESENSE commented on examples/i2c_direct/include/i2c_bmi270.h:
```diff
@@ -0,0 +1,84 @@
+
+int get_bmi270_config_file_size(void);
+uint8_t *get_bmi270_config_file_addr(void);
+
```
```suggestion

#if defined(__cplusplus)
}
#endif

```

SPRESENSE commented on examples/i2c_direct/include/i2c_common.h:
```diff
@@ -0,0 +1,55 @@
+  uint8_t reg, uint8_t *pvalue, int len);
+int i2c_reg_read(i2c_ctrl_t *pi2c,
+  uint8_t reg, uint8_t *value, int16_t len);
+
```
```suggestion

#if defined(__cplusplus)
}
#endif

```

SPRESENSE commented on examples/i2c_direct/i2c_direct_main.c:
```diff
@@ -0,0 +1,298 @@
+ ****************************************************************************/
+
+/**
+ * @brief main - init BMI270 and mesurement 100 times.
```
```suggestion
 * @brief main - init BMI270 and measurement 512 times.
```

SPRESENSE commented on examples/i2c_direct/i2c_direct_main.c:
```diff
@@ -0,0 +1,298 @@
+ * Private Function Prototypes
+ ****************************************************************************/
+
+static void gpio_setup_direction (int id, bool input);
```
```suggestion
static void gpio_setup_direction(int id, bool input);
```

SPRESENSE commented:
@shino-ixy  Remove some warnings.
```
i2c_bmi270.c: In function 'bmi270_fifo_decoder':
i2c_bmi270.c:244:10: warning: 'return' with a value, in function returning void [-Wreturn-type]
  244 |   return 0;
      |          ^
i2c_bmi270.c:161:1: note: declared here
  161 | bmi270_fifo_decoder
      | ^~~~~~~~~~~~~~~~~~~
i2c_bmi270.c: In function 'init_bmi270':
i2c_bmi270.c:312:20: warning: assignment to 'axis_t *' {aka 'struct _axis_type *'} from incompatible pointer type 'uint8_t *' {aka 'unsigned char *'} [-Wincompatible-pointer-types]
  312 |   pctrl->acc_table = (uint8_t *)malloc(BMI270_STORE_TABLE_LENGTH);
      |                    ^
i2c_bmi270.c:322:20: warning: assignment to 'axis_t *' {aka 'struct _axis_type *'} from incompatible pointer type 'uint8_t *' {aka 'unsigned char *'} [-Wincompatible-pointer-types]
  322 |   pctrl->gyr_table = (uint8_t *)malloc(BMI270_STORE_TABLE_LENGTH);
      |                    ^
i2c_bmi270.c: At top level:
i2c_bmi270.c:108:15: warning: array 'bmi270_config_file' assumed to have one element
  108 | const uint8_t bmi270_config_file[];
      |               ^~~~~~~~~~~~~~~~~~
bmi270.c: In function 'get_bmi270_config_file_addr':
bmi270.c:947:10: warning: return discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  947 |   return bmi270_config_file;
      |          ^~~~~~~~~~~~~~~~~~
```

shino-ixy commented:
I’m so sorry for repeated reviewing.

I've updated comments and codes, resolved any issues. these updates are included in single commit.

SPRESENSE commented on examples/i2c_direct/include/i2c_bmi270.h:
```diff
@@ -0,0 +1,97 @@
+#include "debug_printf.h"
+
+/****************************************************************************
+ * Public Types
```
```suggestion
 * Pre-processor Definitions
```

SPRESENSE reviewed (changes requested):
@shino-ixy , thanks for your quick response.
I added a comment and please fix `i2c_bmi270.h`.

shino-ixy commented:
Thank you for checking, I've updated i2c_bmi270.h.

SPRESENSE commented:
@shino-ixy san, could you please add sdk/configs/examples/i2c_direct/README.txt?
Thanks.

shino-ixy commented:
OK! I've just written README.txt, Is this content sufficient?

SPRESENSE commented:
@shino-ixy san
README.txt you added is very helpful. Thank you.

And, please refer to [README.txt](https://github.com/sonydevworld/spresense/blob/develop/sdk/configs/examples/alarm/README.txt), and add a simple README.txt under sdk/configs/examples/i2c_direct directory.

shino-ixy commented:
@SPRESENSE Thank you for checking, I've just added README in configs.

SPRESENSE commented on sdk/configs/examples/i2c_direct/README.txt:
```diff
@@ -0,0 +1,4 @@
+This configuration contains required options to use gyro example.
```
```suggestion
This configuration contains required options to use i2c_direct example.
```

SPRESENSE commented on sdk/configs/examples/i2c_direct/README.txt:
```diff
@@ -0,0 +1,4 @@
+This configuration contains required options to use gyro example.
+
+[Source path]
+examples/gyro
```
```suggestion
examples/i2c_direct
```

SPRESENSE reviewed (changes requested):
@shino-ixy , please modify an example name as I suggested.

shino-ixy commented:
I'm so sorry...for my wrong commit.

SPRESENSE reviewed (approved):
@shino-ixy san, thank you!!
